generator/internal/tracing_connection_generator.h

```diff
@@ -0,0 +1,63 @@
+// Copyright 2023 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_TRACING_CONNECTION_GENERATOR_H
+#define GOOGLE_CLOUD_CPP_GENERATOR_INTERNAL_TRACING_CONNECTION_GENERATOR_H
+
+#include "google/cloud/status.h"
+#include "generator/internal/printer.h"
+#include "generator/internal/service_code_generator.h"
+#include <google/protobuf/compiler/code_generator.h>
+#include <google/protobuf/descriptor.h>
+#include <map>
+#include <memory>
+#include <string>
+
+namespace google {
+namespace cloud {
+namespace generator_internal {
+
+/**
+ * Generates the header file and cc file for the TracingConnection class for a
+ * particular service.
+ */
+class TracingConnectionGenerator : public ServiceCodeGenerator {
+ public:
+  TracingConnectionGenerator(
+      google::protobuf::ServiceDescriptor const* service_descriptor,
+      VarsDictionary service_vars,
+      std::map<std::string, VarsDictionary> service_method_vars,
+      google::protobuf::compiler::GeneratorContext* context);
+
+  ~TracingConnectionGenerator() override = default;
+
+ private:
+  Status GenerateHeader() override;
+  Status GenerateCc() override;
+
+  static std::string MethodDeclaration(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string AsyncMethodDeclaration(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string MethodDefinition(
+      google::protobuf::MethodDescriptor const& method);
+  static std::string AsyncMethodDefinition(
+      google::protobuf::MethodDescriptor const& method);
```

Comment:
Do these need to be members?

Reply:
No. But it is consistent with what we do for the `ConnectionImpl` https://github.com/googleapis/google-cloud-cpp/blob/9196bdcd6281afb62456b648ca436cdfddaf6241/generator/internal/connection_impl_generator.h#L49-L56

(It's probably the laziness talking, but) I'd rather change neither than both.